### PR TITLE
IBP-2393 manage samples treetable angular

### PR DESCRIPTION
**Pull request by nahuel-soldevilla**
- [x] Upgrade to angular 6 
- [x] Replace tree table implementation using primeng treetable 
- [x] Call to sample list services

### CODE REVIEW

**Comment by darla-leafnode** [comment]
Hi @nahuel-soldevilla - we will have conflict in the samples and sample-list API calls/changes. I made some changes in the API permissions branch related to the URLs of these services.
So it's best to merge our changes together in that main "tech sprint" branch, along with other major-impact changes.
cc : @clarysabel

**Comment by nahuel-soldevilla** [comment]
Thanks for the heads up @darla-leafnode 
Don't worry about the merge, depending on which ticket gets finished first I'll update the corresponding branch accordingly with master.

File: pom.xml [hunk 1]
```diff
@@ -15,7 +15,7 @@
 		<spring.version>3.2.11.RELEASE</spring.version>
 		<spring.security.version>3.2.5.RELEASE</spring.security.version>
 		<modelmapper.version>0.7.3</modelmapper.version>
-		<node.version>v6.9.1</node.version>
+		<node.version>v8.11.3</node.version>
```
Comment (nahuel-soldevilla) [inline pom.xml]: @IntegratedBreedingPlatform/developers 

After this is merged, if you have `node` installed globally please remember to upgrade the version too. I usually do it with `n`:
```shell
sudo n 8.11.3
```

File: src/main/jhipster/src/main/webapp/app/entities/sample/tree-table/sample-tree.service.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    private url = '/Fieldbook/SampleListTreeManager';
+    private bmsapiUrl;
+    private initTreeUrl = `${this.url}/loadInitTree/${this.isFolderOnly}`;
+    private programUID: string;
+
+    constructor(private http: HttpClient) {
+    }
+
+    setCrop(crop: string) {
+        // url: '/bmsapi/sampleLists/' + cropName + '/sampleListFolder?folderName=' + folderName + '&parentId=' + parentFolderId + '&programUUID=' + currentProgramId,
```
Comment (nahuel-soldevilla) [inline src/main/jhipster/src/main/webapp/app/entities/sample/tree-table/sample-tree.service.ts]: Remove

File: src/main/jhipster/src/main/webapp/app/entities/sample/tree-table/sample-tree.service.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        const options = createRequestOption(req);
+        return this.http.get<TreeNode[]>(this.initTreeUrl, { params: options, observe: 'response' })
+            .pipe(map((res: HttpResponse<TreeNode[]>) => this.convertArrayResponse(res)));
+    }
+
+    move(source: string, target: string): Observable<any> {
+        const isCropList = target === 'CROPLISTS';
+        const sourceId = source === 'LISTS' || source === 'CROPLISTS' ? 0 : source;
+        const targetId = target === 'LISTS' || target === 'CROPLISTS' ? 0 : target;
+        const url = `${this.bmsapiUrl}/${sourceId}/move?newParentId=${targetId}&isCropList=${isCropList}&&programUUID=${this.programUID}`;
```
Comment (nahuel-soldevilla) [inline src/main/jhipster/src/main/webapp/app/entities/sample/tree-table/sample-tree.service.ts]: remove double ampersands in urls

File: src/main/jhipster/src/main/webapp/app/entities/sample/tree-table/sample-tree.service.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    private bmsapiUrl;
+    private initTreeUrl = `${this.url}/loadInitTree/${this.isFolderOnly}`;
+    private programUID: string;
+
+    constructor(private http: HttpClient) {
+    }
+
+    setCrop(crop: string) {
+        // url: '/bmsapi/sampleLists/' + cropName + '/sampleListFolder?folderName=' + folderName + '&parentId=' + parentFolderId + '&programUUID=' + currentProgramId,
+        this.bmsapiUrl = `/bmsapi/sampleLists/${crop}/sampleListFolder`;
```
Comment (nahuel-soldevilla) [inline src/main/jhipster/src/main/webapp/app/entities/sample/tree-table/sample-tree.service.ts]: Name this one `url` and rename the other one to `fieldbookUrl`

**Pull request merged**